execute/tokendata/usdc/attestation.go

```diff
@@ -30,15 +45,58 @@ type AttestationStatus struct {
 // Ethereum ->
 //
 //	12 ->
-//	  0 -> AttestationStatus{Error: nil, Data: [32]byte{attestation_data}}
-//	  2 -> AttestationStatus{Error: ErrNotRead, Data: nil}
+//	  0 -> AttestationStatus{Error: nil, Attestation: "ABCDEF", MessageHash: bytes}
+//	  2 -> AttestationStatus{Error: ErrNotRead, Attestation: nil, MessageHash: nil}
 type AttestationClient interface {
 	Attestations(
 		ctx context.Context,
 		msgs map[cciptypes.ChainSelector]map[exectypes.MessageTokenID]reader.MessageHash,
 	) (map[cciptypes.ChainSelector]map[exectypes.MessageTokenID]AttestationStatus, error)
 }
 
+type sequentialAttestationClient struct {
+	client HTTPClient
+	hasher hashutil.Hasher[[32]byte]
+}
+
+//nolint:revive
+func NewAttestationClient(config pluginconfig.USDCCCTPObserverConfig) *sequentialAttestationClient {
+	return &sequentialAttestationClient{
+		client: NewHTTPClient(
+			config.AttestationAPI,
+			config.AttestationAPIInterval.Duration(),
+			config.AttestationAPITimeout.Duration(),
+		),
+		hasher: hashutil.NewKeccak(),
+	}
+}
+
+func (s *sequentialAttestationClient) Attestations(
+	ctx context.Context,
+	msgs map[cciptypes.ChainSelector]map[exectypes.MessageTokenID]reader.MessageHash,
+) (map[cciptypes.ChainSelector]map[exectypes.MessageTokenID]AttestationStatus, error) {
+	outcome := make(map[cciptypes.ChainSelector]map[exectypes.MessageTokenID]AttestationStatus)
+
+	for chainSelector, hashes := range msgs {
+		outcome[chainSelector] = make(map[exectypes.MessageTokenID]AttestationStatus)
+
+		for tokenID, messageHash := range hashes {
+			// TODO sequential processing
```

Comment:
Next PR, I'm optimizing for having working usdc flow asap. We can optimize the code by caching or parallel processing later